### PR TITLE
Add some IPM data sources

### PR DESCRIPTION
**Pull request by gschivley**
I've added two data tables from IPM. Data download is the hardest part - I've added a few more `if` and `elif` statements but didn't do much to make the code more flexible.

### CODE REVIEW

**Comment by zaneselvans** [comment]
Looks like you maybe need to add an `update_datastore --source ipm` line to the Travis setup script.

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ install:
 - cd scripts
 - python update_datastore.py -s eia923 -y 2017
 - python update_datastore.py -s eia860 -y 2017
+- pythong update_datastore.py -s epaipm
```
Comment (zaneselvans) [inline .travis.yml]: Hmm, the pythong. I make that typo a lot too.

**Comment by zaneselvans** [comment]
I think you'll also need to go into `conftest.py` where the fixtures are set up for both the `etl` and `travis_ci` tests, and change the parameters that are used to initialize the test database, otherwise the `epaipm` data source won't be exercised by the tests. Particularly the `pudl_engine` and `pudl_engine_travis_ci` fixtures.

**Comment by zaneselvans** [comment]
@gschivley would you be willing to add another test case to `test/datastore_test.py` to exercise the `epaipm` download? I tried adding the simplest case, but it seems like the `epaipm` source isn't giving the expected kind of output for `pudl.datastore.paths_for_year()` (which seems like it ought to just return the same paths, no matter what year it's given, since the data isn't year dependent).

**Comment by gschivley** [comment]
@zaneselvans Sure, I'll give it a try. I ended up bypassing `paths_for_year()` and adding some custom code in `pudl.datastore.organize()`.

```python
    if source == 'epaipm':
        # downloading .xlsx files, not zip files.
        unzip = False

        fns = list(pc.epaipm_url_ext.values())
        newfiles = [
            os.path.join(tmpdir, f)
            for f in fns
        ]
        destfiles = [
            os.path.join(SETTINGS['epaipm_data_dir'], f)
            for f in fns
        ]
    else:
        newfiles = [os.path.join(tmpdir, os.path.basename(f))
                    for f in paths_for_year(source, year, states)]
        destfiles = paths_for_year(
            source, year, states, file=True, datadir=datadir)
```

Looking back at `pudl.datastore.path()`, maybe I can use it with `year=0`. I think the problem is that other sources in `path()` generally have a single filename, but for IPM I'm downloading a bunch of different excel files. To make it more complicated I might also be pulling some tables from PDF files in the future.

**Comment by zaneselvans** [comment]
Ah, okay. Thus far everything we've pulled down has been zipped into one file, so the other (EIA) multi-spreadsheet datasets have had a single zipfile that can be referred to to confirm that the download succeeded.  Like I said, the datastore stuff needs to be better generalized to work with the full variety of datasets that will be incorporated.

**Comment by zaneselvans** [comment]
@gschivley Another minor issue: even when `pudl.init.init_db()` is called with `epaipm_tables=[]` it appears to go ahead and load those tables. I noticed this when I added that argument to the `test.etl_test.test_only_ferc1_pudl_init_db()` test case (which failed in the full etl tests because the default `None` value wasn't iterable). All the other datasets have an ETL offramp -- where if no years or no tables are requested, they skip ETL entirely.

Unfortunately because our tests use a bunch of data, the Travis CI tests aren't able to run everything, so running the full ETL tests locally at least is a good idea before trying to merge.  i.e.

```sh
pytest --disable-warnings test/etl_test.py
```

**Comment by gschivley** [comment]
@zaneselvans I think that the ETL offramp is fixed. ETL is only performed for tables in the settings yml file.

File: pudl/extract/epaipm.py [hunk 2]
```diff
@@ -26,8 +27,16 @@ def get_epaipm_name(file):
     Returns:
         path to EPA IPM spreadsheet.
     """
+    if sorted(datadir.glob(file)):
+        name = sorted(datadir.glob(file))[0]
+    elif sorted(pudl_datadir.glob(file)):
+        name = name = sorted(pudl_datadir.glob(file))[0]
```
Comment (zaneselvans) [inline pudl/extract/epaipm.py]: Is this double assignment of `name` intentional?

Comment (gschivley) [reply pudl/extract/epaipm.py]: Nope. Thanks for the catch.

**Pull request merged**